File: home/dot_local/lib/theme-gallery/samples/sample.py
"""Palette sample: Python — docstrings, strings, numbers, keywords, types."""
from __future__ import annotations

MAX_RETRIES: int = 3


class Greeter:
    """Builds greetings."""

    def __init__(self, name: str = "world") -> None:
        self.name = name

    def greet(self) -> str:
        parts = [f"hello {self.name} ({i})\n" for i in range(MAX_RETRIES)]
        return "".join(parts)


if __name__ == "__main__":
    print(Greeter("palette").greet())

File: home/dot_local/lib/theme-gallery/samples/sample.lua
-- Palette sample: lua — comments, strings, numbers, keywords, functions.
local M = {}

local MAX = 42
local name = "world\n"

--- Doc-style comment describing the greeter.
function M.greet(who)
  who = who or name
  for i = 1, MAX do
    if i % 7 == 0 then
      print(("hello %s (%d)"):format(who, i))
    end
  end
  return true
end

return M
